src/platform/forms-system/src/js/utilities/data/findDuplicateIndexes.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Find duplicates in an array of objects and return an array of indexes for the
+ *  duplicates only
+ * @param {Object} formData - Data array (ArrayField data)
+ * @param {String} dataKey - key of data of interest to find duplicates
+ * @return {Array} - Array of indexes of the duplicates only, not the first
+ *  entry
+ * @example
+ * findDuplicateIndexes([{x:'one'},{x:'two'},{x:'one'},{x:'two'}], 'x')
+ * // => [2, 3]
+ */
+export default function(formData = [], dataKey = '') {
+  if (!dataKey || formData?.length === 0) {
+    return [];
+  }
+  const list = formData.map(item => item[dataKey]?.toLowerCase() || '');
+  return list.reduce((duplicateIndexes, item, index) => {
+    // look for duplicates of the first, but don't include the first
+    const foundIndex = list.indexOf(item, index + 1);
+    if (foundIndex > -1) {
+      duplicateIndexes.push(foundIndex);
+    }
+    return duplicateIndexes;
+  }, []);
```

Comment:
This is clever. It took me a while to understand why this was less work than mapping over the list to get the indexes, then filtering out the `null`s (non-duplicates). How might we make this easier to understand more quickly?

Specifically, it took a while to realize that this finds all the duplicates for multiple values of `item[dataKey]`. That is, it'll find the duplicate `{ x: 'one' }` _and_ `{ x: 'two' }` in your example.

Reply:
Thanks! I'll add a comment explaining that it looks for a single duplicate of the currently indexed item, so even if there is more than one duplicate of a single item or multiple duplicates of multiple items, it will include them. Sound good?

Reply:
The comment you added looks great. :raised_hands: